src/components/LoginPage/LoginPageMain.jsx

```diff
@@ -0,0 +1,138 @@
+import React, { useState } from "react";
+import "../LoginPage/LoginPage.css";
+import email_image from "../assets/email.png";
+import person_image from "../assets/person.png";
+import password_image from "../assets/password.png";
+
+const LoginPageMain = () => {
+  const [action, setAction] = useState("Login");
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [name, setName] = useState("");
+  const [loginError, setLoginError] = useState("");
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+
+  const [users, setUsers] = useState([]);
+
+  const handleLogin = () => {
+    const user = users.find(
+      (user) => user.email === email && user.password === password
+    );
+
+    if (user) {
+      setIsLoggedIn(true);
+      setLoginError("");
+    } else {
+      setLoginError("Invalid email or password");
+    }
+  };
+
+  const handleSignUp = () => {
+    if (users.find((user) => user.email === email)) {
+      setLoginError("User with this email already exists.");
+      return;
+    }
+
+    if (name && email && password) {
+      setUsers([...users, { name, email, password }]);
```

Comment:
I think you put this in for testing, but we don't need to store a local `users` state variable here since it should get replaced very soon with an actual API call.

See the comment below about separating login and signup components - this will make this state variable shared between the two much more complicated to keep around, so I think we can just get rid of this code.